update.sh

```diff
@@ -33,23 +33,24 @@ hour=$(echo $hour | sed 's/^0*//')
 # Different desktop environment implementations
 case $XDG_CURRENT_DESKTOP in
 	Mint|Mate) setcmd="gsettings set org.mate.background picture-uri";;
-	Cinnamon) setcmd="gsettings set org.cinnamon.background picture-uri";;
-	*) setcmd="gsettings set org.gnome.desktop.background picture-uri";; # GNOME/Unity, default
+	MATE) setcmd="gsettings set org.mate.background picture-filename ";;
+	Cinnamon) setcmd="gsettings set org.cinnamon.background picture-uri file://";;
+	*) setcmd="gsettings set org.gnome.desktop.background picture-uri file://";; # GNOME/Unity, default
```

Comment:
Can you confirm your release always uses caps in `$XDG_CURRENT_DESKTOP`? Seeing as how the previous line handling Mate was left intact, we just need to know it wouldn't cause a weird conflict. If this value is version-specific, we might want to consider a better check than this, and if not, the previous regex for `Mate` might be unnecessary